File: mobile/lib/features/auth/models/login_response.dart
class LoginResponseModel {
  LoginResponseModel({
    required this.appUserId,
    required this.username,
    required this.accessToken,
    required this.expiresAtUtc,
    required this.role,
  });

  final int appUserId;
  final String username;
  final String accessToken;
  final DateTime expiresAtUtc;
  final String role;

  factory LoginResponseModel.fromJson(Map<String, dynamic> json) {
    return LoginResponseModel(
      appUserId: json['appUserId'] as int,
      username: json['username'] as String,
      accessToken: json['accessToken'] as String,
      expiresAtUtc: DateTime.parse(json['expiresAtUtc'] as String),
      role: json['role'] as String,
    );
  }
}

File: backend/src/Contracts/Auth/LoginResponse.cs
namespace OptimaHealthcare.Contracts.Auth;

public sealed class LoginResponse
{
    public required int AppUserId { get; init; }
    public required string Username { get; init; }
    public required string AccessToken { get; init; }
    public required DateTime ExpiresAtUtc { get; init; }
    public required string Role { get; init; }
}
